src/blocks/weather.rs

```diff
@@ -215,20 +284,43 @@ pub async fn run(config: &Config, api: &CommonApi) -> Result<()> {
             None
         };
 
-        let fetch = || provider.get_weather(location);
+        let fetch = || provider.get_weather(location, need_forecast(&format, &format_alt));
```

Comment:
Not that it is particularly expensive, but `need_forecast(&format, &format_alt)` can be computed once outside the loop.